src/main/java/minicraft/core/io/FileHandler.java

```diff
@@ -159,38 +165,28 @@ public FileVisitResult visitFileFailed(Path p, IOException ex) {
 			deleteFolder(origFolder.toFile());
 	}
 
-	/** https://stackoverflow.com/questions/1429172/how-do-i-list-the-files-inside-a-jar-file/1429275#1429275 */
-	public static ArrayList<String> listResources() {
-		ArrayList<String> names = new ArrayList<>();
+	public static ArrayList<String> listAssets() {
+		Path path;
 		try {
-			CodeSource src = Game.class.getProtectionDomain().getCodeSource();
-			if (src != null) {
-				URL jar = src.getLocation();
-				ZipInputStream zip = new ZipInputStream(jar.openStream());
-				int reads = 0;
-				while (zip.available() == 1) {
-					ZipEntry e = zip.getNextEntry();
-
-					// e is either null if there are no entries left, or if
-					// we're running this from an ide
-					if (e == null) {
-						if (reads > 0) break;
-						else {
-							return listResourcesUsingIDE();
-						}
-					}
-					reads++;
-					names.add(e.getName());
-				}
-			} else {
-				Logging.RESOURCEHANDLER_LOCALIZATION.error("Failed to get code source.");
-				return names;
+			path = Paths.get(Objects.requireNonNull(Game.class.getResource("/assets/")).toURI());
+		} catch (URISyntaxException e) {
+			throw new RuntimeException(e); // FATAL ERROR
+		} catch (FileSystemNotFoundException e) {
+			try {
+				FileSystem fs = FileSystems.newFileSystem(Objects.requireNonNull(Game.class.getResource("/assets/")).toURI(), Collections.emptyMap());
+				path = fs.getPath("/assets/");
+			} catch (URISyntaxException | IOException ee1) {
+				throw new RuntimeException(ee1); // FATAL ERROR
 			}
+		}
 
+		ArrayList<String> names = new ArrayList<>();
+		try (Stream<Path> paths = Files.walk(path)) {
+			Path finalPath = path;
+			paths.forEach(p -> names.add(finalPath.getParent().relativize(p).toString()));
 			return names;
 		} catch (IOException e) {
-			CrashHandler.errorHandle(e);
-			return names;
+			throw new RuntimeException(e); // FATAL ERROR
```

Comment:
A fatal error should probably stop the game. That's what fatal means.

Reply:
Currently, there is no crash report system, only popups are used. This error is thrown, unhandled error is caught by the handler.